spring-batch-integration/src/main/java/org/springframework/batch/integration/chunk/RemoteChunkingMasterStepBuilderFactory.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.batch.integration.chunk;
+
+import org.springframework.batch.core.repository.JobRepository;
+import org.springframework.context.support.GenericApplicationContext;
+import org.springframework.transaction.PlatformTransactionManager;
+
+/**
+ * Convenient factory for a {@link RemoteChunkingMasterStepBuilder} which sets
+ * the {@link JobRepository}, {@link PlatformTransactionManager} and
+ * {@link GenericApplicationContext} automatically.
+ *
+ * @since 4.1
+ * @author Mahmoud Ben Hassine
+ */
+public class RemoteChunkingMasterStepBuilderFactory {
```

Comment:
Is there a reason to expose this or should we make this an inner class to the `BatchIntegrationConfiguration`?

Reply:
I used the same approach as `StepBuilderFactory` and `JobBuilderFactory` which are public. It could be an inner class of `BatchIntegrationConfiguration`, but in this case, when we autowire the beans, we would write something like:

```java
@Autowired
private BatchIntegrationConfiguration.RemoteChunkingMasterStepBuilderFactory factory;
```

IMO, the user has not to know about the "internal" `BatchIntegrationConfiguration` (even if it's public) just like `AbstractBatchConfiguration` and its subclasses.